src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -0,0 +1,305 @@
+        {
+            // If the context is missing, stop
+            if (context == null)
+            {
+                return null;
+            }
+
+            var bufferDictionary = GetBufferDictionary(context);
+
+            // If the dictionary is missing, stop
+            if (bufferDictionary == null)
+            {
+                return null;
+            }
+
+            lock (bufferDictionary)
+            {
+                if (!bufferDictionary.TryGetValue(this, out var buffer))
+                {
+                    buffer = new Internal.LogEventInfoBuffer(BufferSize, GrowBufferAsNeeded, BufferGrowLimit);
+                    bufferDictionary.Add(this, buffer);
+                }
+
+                return buffer;
+            }
+        }
+
+        /// <summary>
+        /// The Key for the buffer dictionary in the HttpContext.Items collection
+        /// </summary>
+        private static readonly object HttpContextItemsKey = new object();
+
+        /// <summary>
+        /// Return the buffer dictionary from the HttpContext.Items using HttpContextItemsKey
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private static Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer> GetBufferDictionary(HttpContextBase context)
+        {
+            return context?.Items?[HttpContextItemsKey] as
+                Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer>;
+        }
+
+        /// <summary>
+        /// Create the buffer dictionary in the HttpContext.Items using HttpContextItemsKey
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private static void SetBufferDictionary(HttpContextBase context)
+        {
+            context.Items[HttpContextItemsKey] = new Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer>();
+        }
+
+        /// <summary>
+        /// Adds the specified log event to the buffer.
+        /// NOTE: if Write is never called, this instance will not be registered in the buffer dictionary in HttpContext.Items.
+        /// That is expected normal behavior.
+        /// </summary>
+        /// <param name="logEvent">The log event.</param>
+        protected override void Write(AsyncLogEventInfo logEvent)
+        {
+            var buffer = GetOrCreateRequestBuffer(HttpContextAccessor.HttpContext);
+            if (buffer != null)
+            {
+                WrappedTarget?.PrecalculateVolatileLayouts(logEvent.LogEvent);
+                InternalLogger.Trace("Appending log event {0} to ASP.NET request buffer.", logEvent.LogEvent.SequenceID);
+                buffer.Append(logEvent);
+            }
+            else
+            {
+                InternalLogger.Trace("ASP.NET request buffer does not exist. Passing to wrapped target.");
+                WrappedTarget?.WriteAsyncLogEvent(logEvent);
```

Comment:
Can `WrappedTarget?` be changed to pure `WrappedTarget` ? (no conditional)

I know this is not optimal, but we need to reduce the changes to make the rename automatic